proto/redpanda/api/dataplane/v1alpha1/acl.proto

```diff
@@ -93,13 +93,87 @@ message ListACLsResponse {
 }
 
 message CreateACLRequest {
-  ACL.ResourceType resource_type = 1;
+  option (buf.validate.message).cel = {
+    id: "resource_name_must_be_set_except_for_cluster_resource_type",
+    expression:
+      "this.resource_type == 4 && size(this.resource_name) == 0 ? ''"
```

Comment:
Yes we could, but what I didn't like is that you have to provide the full package path to the enum. So it wouldn't be `this.resource_type == RESOURCE_TYPE_CLUSTER` it would be: `this.resource_type == redpanda.dataplane.v1alpha1.ACL.ResourceType.RESOURCE_TYPE_CLUSTER`. Since you have almost no validation at compile time, nor a good IDE support in the form of intellisense for this I did not want to risk runtime CEL errors and used the plain numbers, which seem less error-prone to me.